### PR TITLE
Feat/upgrade purgecss to version 4

### PR DESCRIPTION
**Pull request by dogpatch626**
BREAKING CHANGE: various whitelist options replaced with safelist

## Description
Upgrade purgecss to version 4.

## Motivation and Context

Fixes https://github.com/americanexpress/purgecss-loader/issues/54

Release notes for purgecss 3, which describe the change from whitelist to safelist:
https://github.com/FullHuman/purgecss/releases/tag/v3.0.0

Release notes for purgecss 4, which describe the drop of support for postcss4:
https://github.com/FullHuman/purgecss/releases/tag/v4.0.0

## How Has This Been Tested?
Updated unit tests to pass.

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation (adding or updating documentation)
- [x] Dependency update

## Checklist:

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
- [x] My changes are in sync with the code style of this project.
- [x] There aren't any other open Pull Requests for the same issue/update.
- [ ] These changes should be applied to a maintenance branch.
- [ ] I have added the Apache 2.0 license header to any new files created.

## What is the Impact to Developers Using Purgecss-Loader?
This resolves a security issue.  Developers will need to update their configurations.

### CODE REVIEW

File: americanexpress-purgecss-loader-3.0.0.tgz
Comment (Matthew-Mallimo) [inline americanexpress-purgecss-loader-3.0.0.tgz:1]: Lets remove this, and add `.tgz` to the .gitignore

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 node_modules
 test-results
 .jest-cache
+.tgz
```
Comment (Matthew-Mallimo) [inline .gitignore:8]:
```suggestion
*.tgz
```

Comment (dogpatch626) [reply .gitignore]: thanks Matt didn't realize i needed to glob that

**Review by Matthew-Mallimo — APPROVED** [review]
Can you rename the PR to `upgrade purgecss to version 4, since we are using v4 :)

**Pull request merged**